addons/l10n_latam_check/tests/test_third_party_checks.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.addons.l10n_latam_check.tests.common import L10nLatamCheckTest
+from odoo.exceptions import ValidationError, UserError
+from odoo.tests.common import tagged
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestThirdChecks(L10nLatamCheckTest):
+
+    def create_third_party_check(self, journal=False):
+        if not journal:
+            journal = self.third_party_check_journal
+        vals = {
+            'partner_id': self.partner_a.id,
+            'amount': '00000001',
+            'check_number': '00000001',
+            'payment_type': 'inbound',
+            'journal_id': journal.id,
+            'payment_method_line_id': self.third_party_check_journal._get_available_payment_method_lines('inbound').filtered(lambda x: x.code == 'new_third_party_checks').id,
+        }
+        payment = self.env['account.payment'].create(vals)
+        payment.action_post()
+        return payment
+
+    def test_01_get_paid_with_multiple_checks(self):
+        """ This a generic test to check that we are able to paid with checks
+
+        We paid directly with multiple checks insteaf of one just check, just to ensure the create multi
+        is properly working (we have a bug before because it was failing) """
+        vals_list = [{
+            'partner_id': self.partner_a.id,
+            'amount': '00000001',
+            'check_number': '00000001',
+            'payment_type': 'inbound',
+            'journal_id': self.third_party_check_journal.id,
+            'payment_method_line_id': self.third_party_check_journal._get_available_payment_method_lines('inbound').filtered(lambda x: x.code == 'new_third_party_checks').id,
+        }, {
+            'partner_id': self.partner_a.id,
+            'amount': '00000002',
+            'check_number': '00000002',
+            'payment_type': 'inbound',
+            'journal_id': self.third_party_check_journal.id,
+            'payment_method_line_id': self.third_party_check_journal._get_available_payment_method_lines('inbound').filtered(lambda x: x.code == 'new_third_party_checks').id,
+        }]
+        payments = self.env['account.payment'].create(vals_list)
+        payments.action_post()
+        self.assertEqual(len(payments), 2, 'Checks where not created properly')
+        for payment in payments:
+            self.assertEqual(payment.state, 'posted', 'Check %s was not created properly' % payment.check_number)
+            self.assertEqual(payment.l10n_latam_check_current_journal_id, self.third_party_check_journal, 'Current journal was not computed properly')
+
+    def test_02_third_party_check_delivery(self):
```

Comment:
William, as we are not changing how the payment / journal entries / items are being create, I believe that we can only test this.

Actually maybe whe can remove the assert of payment.state and only assert of "payment.l10n_latam_check_current_journal_id" as that behaviour is the once introduced by this module. 

The module is not changing anything related to accounting / payments functionality.

Ok to keep as it is?

@william-andre

Reply:
Checking the balance is the only way of knowing that everything is right.
We are dealing with payments in multiple journals, referencing each other, cancelling each other, etc.

Checking the state everywhere is useless indeed though. Just not having an error should be enough.

Reply:
@william-andre @jjscarafia @zaoral can we close this comment?
cc' @jco-odoo

Reply:
We have removed the check of the state everywhere as suggested, because just not having an error posting the payment should be enough. commit d2fe80b8e914e95fe5d35e80534e372e5131b45e

Reply:
Mmmh, ok for not checking the balance...